orc8r/gateway/python/magma/magmad/config_manager.py

```diff
@@ -145,4 +175,16 @@ def did_mconfig_change(serv_name):
             if srv in mconfig.configs_by_key:
                 configs_by_key[srv] = mconfig.configs_by_key.get(srv)
 
+            agw_version = self._magmad_service.version
+            unpacked_mconfig = mconfig.configs_by_key.get(MAGMAD)
+            version_info = self._warn_if_versions_are_incompatible(agw_version, unpacked_mconfig)
+            agw_version_parsed = version_info.agw_version
+            orc8r_version_parsed = version_info.orc8r_version
+
+            if agw_version_parsed and orc8r_version_parsed:
+                agw_minor = int(agw_version_parsed.group('minor_version'))
+                orc8r_minor = int(orc8r_version_parsed.group('minor_version'))
+                if agw_minor - orc8r_minor <= -1:
+                    logging.warning("Gateway is more than one minor version behind orc8r. Please consider updating it.")
```

Comment:
Should we just pull these into `_warn_if_versions_are_incompatible`?  Seems cleaner that way.
Or rename `_warn_if_versions_are_incompatible` -> `_parse_version_info_from_mconfig`?

Reply:
`_warn_if_versions_are_incompatible` is quite descriptive ,but i think naming is very hard, so if it works then its fine.

Reply:
Yes, I think the naming is good, but it is a bit confusing if we have logic to log warnings about versions inside and outside of the function. So we should have a single function that includes all version warning logs, or split them up into smaller logics with descriptive names.

Reply:
or we can name it more descriptively. i dont agree with breaking it further into smaller function, i think this function represents a very tightly couple functionality. it is parsing the versions it received, and logs its way through its processing steps, to make it easier for people to debug, its actual purpose is not to log but to parse, and log messages are there to assist with debugging a future problem.

Reply:
Yep renaming sounds good!